app/experimenter/docs/adrs/0004-monitoring-data-import.md

```diff
@@ -0,0 +1,92 @@
+
+* Pros:
+  * No additional complexity for data processing in Experimenter
+  * Allows flexibility for how data is visualized in Experimenter
+  * Data will load faster coming from a bucket than needing to wait for a BigQuery query to run
+* Cons
+  * Since, so far, the demand on Grafana is low, this option to run the full ETL queries every 10 minutes or so may be much more expensive than option (2) or (3) 
+
+#### (2) Query Data On-Demand from Experimenter
+Add a background celery task that fetches data on-demand from bigquery and stores it in Redis. Redis cache can be used for the next fetch of the same data, but it gets invalidated periodically.
+
+* Pros:
+  * It won't be any slower than the current Grafana graphs
+  * Allows flexibility for how data is visualized in Experimenter
+  * The caching layer would make this approach the same price or cheaper than (1) and (3)
+* Cons
+  * This may be a little slow from the user's perspective compared to having the data computed in the ETL in advance (1)
+  * Requires building complex data fetching, processing and storage logic directly in Experimenter, which means more tests and maintenance burden
+  
+#### (3) Embed Existing Grafana Graphs
+Embed existing Grafana graphs (e.g. [graphs](https://grafana.telemetry.mozilla.org/d/XspgvdxZz/experiment-enrollment?orgId=1&var-experiment_id=bug-1656561-rapid-testing-demo-nightly-80&from=1596153600000&to=1598832000000))
+
+* Pros:
+  * The simplest and quickest solution
+* Cons
+  * Does not allow for flexibility in how data is visualized in Experimenter
+  * Slow compared to (1)
+  * Could be more expensive than (2) unless a caching layer is added
+
+
+#### (4) Option (1) + ETL Optimizations
+Option (1) would be the ideal solution if it was not the most expensive. This ADR initiated ideas about ETL optimizations for option (1) to be implemented with much lower costs.
+
+The idea is to add an hourly job that looks almost exactly like the [daily job](https://github.com/mozilla/bigquery-etl/blob/master/sql/moz-fx-data-shared-prod/telemetry_derived/experiment_enrollment_aggregates_v1/query.sql) that already runs to aggregate 5-minute intervals of live data. Instead, however, the filter would be for 1 hour instead of 1 day and it would operate on the live table instead of `telemetry_derived.event_events_v1` (which is only updated daily). It would append to a table once per hour with the 5-min interval aggregations. This new table would be included as a third table in the [union for `all_enrollments`](https://github.com/mozilla/bigquery-etl/blob/master/sql/moz-fx-data-shared-prod/telemetry_derived/experiment_enrollment_aggregates_live/view.sql.py#L80) in [this view](https://github.com/mozilla/bigquery-etl/blob/master/sql/moz-fx-data-shared-prod/telemetry_derived/experiment_enrollment_aggregates_live/view.sql.p).
```

Comment:
Should we mention the lag here? I'm looking for the docs where this lives, but IIRC the pipeline's SLA is 15 minutes in lag for processing